backend/endpoints/rom.py

```diff
@@ -1337,3 +1336,133 @@ async def get_romfile_content(
     return FileRedirectResponse(
         download_path=Path(f"/library/{file.full_path}"),
     )
+
+
+DEFAULT_PUBLIC_ONLY = Query(False, description="Only return public notes")
+DEFAULT_SEARCH = Query(None, description="Search notes by title or content")
+DEFAULT_TAGS = Query(None, description="Filter by tags")
+
+
+@protected_route(
+    router.get,
+    "/{id}/notes",
+    [Scope.ROMS_READ],
+    responses={status.HTTP_404_NOT_FOUND: {}},
+)
+async def get_rom_notes(
+    request: Request,
+    id: Annotated[int, PathVar(description="Rom internal id.", ge=1)],
+    public_only: bool = DEFAULT_PUBLIC_ONLY,
+    search: str = DEFAULT_SEARCH,
+    tags: list[str] = DEFAULT_TAGS,
+) -> list[UserNoteSchema]:
+    """Get all notes for a ROM."""
+    from handler.database import db_rom_handler
+
+    rom = db_rom_handler.get_rom(id)
+    if not rom:
+        raise RomNotFoundInDatabaseException(id)
+
+    if tags is None:
+        tags = []
+
+    notes = db_rom_handler.get_rom_notes(
+        rom_id=id,
+        user_id=request.user.id,
+        public_only=public_only,
+        search=search,
+        tags=tags,
+    )
+
+    return [UserNoteSchema.model_validate(note) for note in notes]
+
+
+@protected_route(
+    router.post,
+    "/{id}/notes",
+    [Scope.ROMS_USER_WRITE],
+    responses={status.HTTP_404_NOT_FOUND: {}},
+)
+async def create_rom_note(
+    request: Request,
+    id: Annotated[int, PathVar(description="Rom internal id.", ge=1)],
+    note_data: Annotated[dict, Body()],
+) -> UserNoteSchema:
+    """Create a new note for a ROM."""
+    from handler.database import db_rom_handler
+
+    rom = db_rom_handler.get_rom(id)
+    if not rom:
+        raise RomNotFoundInDatabaseException(id)
+
+    note = db_rom_handler.create_rom_note(
+        rom_id=id,
+        user_id=request.user.id,
+        title=note_data["title"],
+        content=note_data.get("content", ""),
+        is_public=note_data.get("is_public", False),
+        tags=note_data.get("tags", []),
+    )
+
+    # Add username to the note data
+    note["username"] = request.user.username
+    return UserNoteSchema.model_validate(note)
+
+
+@protected_route(
+    router.put,
+    "/{id}/notes/{note_id}",
+    [Scope.ROMS_USER_WRITE],
+    responses={status.HTTP_404_NOT_FOUND: {}},
+)
+async def update_rom_note(
+    request: Request,
+    id: Annotated[int, PathVar(description="Rom internal id.", ge=1)],
+    note_id: Annotated[int, PathVar(description="Note id.", ge=1)],
+    note_data: Annotated[dict, Body()],
+) -> UserNoteSchema:
+    """Update a ROM note."""
+    from handler.database import db_rom_handler
+
+    note = db_rom_handler.update_rom_note(
```

Comment:
permission checks should also be at the route level for get/update/delete endpoints